File: apps/HealthApp/android/app/src/main/java/com/myhealth/app/ui/care/CarePlanCard.kt
package com.myhealth.app.ui.care

import androidx.compose.foundation.background
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.myhealth.app.ui.theme.CarePlusColor

/**
 * Per-condition Care plan card. Mirrors iOS `CarePlanCard.swift`.
 *  • title + reading pill
 *  • goal sentence
 *  • intervention tags
 */
@Composable
fun CarePlanCard(
    title: String,
    goal: String,
    interventions: List<String>,
    reading: String? = null,
    readingHealthy: Boolean = true,
) {
    val tint = CarePlusColor.CareBlue
    val pillColor = if (readingHealthy) CarePlusColor.Success else CarePlusColor.Warning

    Card(
        modifier = Modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(MaterialTheme.colorScheme.surfaceVariant),
        shape = RoundedCornerShape(12.dp)
    ) {
        Column(Modifier.padding(12.dp), verticalArrangement = Arrangement.spacedBy(6.dp)) {
            Row {
                Text(title, fontWeight = FontWeight.Bold, modifier = Modifier.weight(1f))
                if (reading != null) {
                    Box(
                        Modifier.background(pillColor.copy(alpha = 0.18f), RoundedCornerShape(99.dp))
                            .padding(horizontal = 10.dp, vertical = 4.dp)
                    ) { Text(reading, color = pillColor, fontWeight = FontWeight.SemiBold,
                        fontSize = 13.sp) }
                }
            }
            Text(goal, color = MaterialTheme.colorScheme.onSurfaceVariant, fontSize = 12.sp)
            if (interventions.isNotEmpty()) {
                val scroll = rememberScrollState()
                Row(Modifier.horizontalScroll(scroll),
                    horizontalArrangement = Arrangement.spacedBy(6.dp)) {
                    interventions.forEach { tag ->
                        Box(
                            Modifier.background(tint.copy(alpha = 0.10f),
                                RoundedCornerShape(99.dp))
                                .padding(horizontal = 8.dp, vertical = 3.dp)
                        ) { Text(tag, color = tint, fontWeight = FontWeight.SemiBold,
                            fontSize = 11.sp) }
                    }
                }
            }
        }
    }
}

/** Per-condition copy. Mirrors iOS [CarePlanRecipe.swift]. */
object CarePlanRecipe {
    data class Recipe(val title: String, val goal: String, val interventions: List<String>)

    fun recipe(condition: String): Recipe = when (condition.lowercase()) {
        "hypertension" -> Recipe("Hypertension", "Goal: under 130/80",
            listOf("Low sodium", "BP log", "DASH plan"))
        "lowbloodpressure" -> Recipe("Low blood pressure",
            "Stay hydrated; avoid sudden standing", listOf("Hydration", "Salt"))
        "heartcondition" -> Recipe("Heart condition", "Stay under prescribed HR cap",
            listOf("HR cap", "Rest day"))
        "diabetest1" -> Recipe("Type 1 diabetes", "Pre/post-meal glucose check",
            listOf("Carb count", "Insulin log"))
        "diabetest2" -> Recipe("Type 2 / Prediabetes",
            "Goal: A1C under 5.7 by next labs",
            listOf("Diet", "Exercise", "Lab retest"))
        "obesity" -> Recipe("Weight management",
            "Steady cardio + 0.5 kg/week deficit", listOf("DASH", "Walk 10k"))
        "asthma" -> Recipe("Asthma", "Carry inhaler; watch AQI",
            listOf("AQI alert", "Inhaler log"))
        "kneeinjury", "ankleinjury", "shoulderinjury", "backpain", "osteoporosis" ->
            Recipe(condition.replaceFirstChar(Char::titlecase),
                "Avoid high-impact; mobility work daily", listOf("Mobility", "Ice/heat"))
        "kidneyissue" -> Recipe("Kidney (CKD)", "Low-K and low-P meals",
            listOf("Low K", "Low P"))
        "anemia" -> Recipe("Anemia", "Iron + vitamin-C pairings",
            listOf("Iron", "Vit C"))
        else -> Recipe(condition.replaceFirstChar(Char::titlecase),
            "Tap to see condition-specific guidance.", emptyList())
    }

    /** Stub readings — wired to Health Connect in week 2. */
    fun readingFor(condition: String): Pair<String, Boolean>? = when (condition.lowercase()) {
        "hypertension" -> "138/88" to false
        "diabetest2", "diabetest1" -> "A1C 6.1" to false
        "heartcondition" -> "HR 72" to true
        "obesity" -> "BMI 31" to false
        else -> null
    }
}

File: apps/HealthApp/ios/FitFusion/Views/Care/CarePlanCard.swift
import SwiftUI
import FitFusionCore

/// Per-condition care plan card. Mirrors the design-spec mockup:
///  • Title + symbol on the left
///  • Latest reading pill on the right (color-coded)
///  • Goal line beneath
///  • Intervention tags as small pills at the bottom
///
/// Reading values come from a `CarePlanRecipe` map (see below) which
/// week-1 builds against the on-device `HealthConditionsStore`. Real
/// values are populated by the new `CarePlanReadings` helper, which in
/// week-2 will read HealthKit (BP, glucose, HRV) directly. For now
/// readings are placeholders unless the caller hands in a value.
public struct CarePlanCard: View {
    public let condition: HealthCondition
    public let reading: String?
    public let readingHealthy: Bool

    private let tint = CarePlusPalette.careBlue

    public init(condition: HealthCondition, reading: String? = nil, readingHealthy: Bool = true) {
        self.condition = condition
        self.reading = reading
        self.readingHealthy = readingHealthy
    }

    public var body: some View {
        let recipe = CarePlanRecipe.recipe(for: condition)
        VStack(alignment: .leading, spacing: 8) {
            HStack(alignment: .top) {
                HStack(spacing: 8) {
                    Image(systemName: condition.symbol).foregroundStyle(tint)
                    Text(recipe.title).font(.headline)
                }
                Spacer()
                if let r = reading {
                    Text(r).font(.subheadline.weight(.semibold))
                        .padding(.horizontal, 10).padding(.vertical, 4)
                        .background(
                            (readingHealthy ? CarePlusPalette.success
                                            : CarePlusPalette.warning).opacity(0.18),
                            in: Capsule()
                        )
                        .foregroundStyle(readingHealthy ? CarePlusPalette.success
                                                         : CarePlusPalette.warning)
                }
            }
            Text(recipe.goal).font(.caption).foregroundStyle(.secondary)
            if !recipe.interventions.isEmpty {
                ScrollView(.horizontal, showsIndicators: false) {
                    HStack(spacing: 6) {
                        ForEach(recipe.interventions, id: \.self) { tag in
                            Text(tag).font(.caption2.weight(.semibold))
                                .padding(.horizontal, 8).padding(.vertical, 3)
                                .background(tint.opacity(0.10), in: Capsule())
                                .foregroundStyle(tint)
                        }
                    }
                }
            }
        }
        .padding(CarePlusSpacing.md)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(CarePlusPalette.surfaceElevated,
                    in: RoundedRectangle(cornerRadius: CarePlusRadius.md))
    }
}

/// Per-condition copy: human-readable title, goal sentence, list of
/// intervention tags. Stored here so a designer can iterate on the
/// language without touching the rendering view.
public enum CarePlanRecipe {
    public struct Recipe {
        public let title: String
        public let goal: String
        public let interventions: [String]
    }

    public static func recipe(for c: HealthCondition) -> Recipe {
        switch c {
        case .hypertension:
            return Recipe(title: "Hypertension",
                          goal: "Goal: under 130/80",
                          interventions: ["Low sodium", "BP log", "DASH plan"])
        case .lowBloodPressure:
            return Recipe(title: "Low blood pressure",
                          goal: "Stay hydrated; avoid sudden standing",
                          interventions: ["Hydration", "Salt"])
        case .heartCondition:
            return Recipe(title: "Heart condition",
                          goal: "Stay under prescribed HR cap",
                          interventions: ["HR cap", "Rest day"])
        case .diabetesT1:
            return Recipe(title: "Type 1 diabetes",
                          goal: "Pre/post-meal glucose check",
                          interventions: ["Carb count", "Insulin log"])
        case .diabetesT2:
            return Recipe(title: "Type 2 diabetes / Prediabetes",
                          goal: "Goal: A1C under 5.7 by next labs",
                          interventions: ["Diet", "Exercise", "Lab retest"])
        case .obesity:
            return Recipe(title: "Weight management",
                          goal: "Steady cardio + 0.5 kg/week deficit",
                          interventions: ["DASH", "Walk 10k"])
        case .asthma:
            return Recipe(title: "Asthma",
                          goal: "Carry inhaler; watch AQI",
                          interventions: ["AQI alert", "Inhaler log"])
        case .pregnancy:
            return Recipe(title: "Pregnancy",
                          goal: "Folate, iron, low-impact cardio",
                          interventions: ["Folate", "Walk", "Yoga"])
        case .kneeInjury, .ankleInjury, .shoulderInjury, .backPain, .osteoporosis:
            return Recipe(title: c.label,
                          goal: "Avoid high-impact; mobility work daily",
                          interventions: ["Mobility", "Ice/heat"])
        case .kidneyIssue:
            return Recipe(title: "Kidney (CKD)",
                          goal: "Low-K and low-P meals",
                          interventions: ["Low K", "Low P"])
        case .liverIssue:
            return Recipe(title: "Liver",
                          goal: "Low alcohol, watch acetaminophen",
                          interventions: ["No alcohol"])
        case .anemia:
            return Recipe(title: "Anemia",
                          goal: "Iron + vitamin-C pairings",
                          interventions: ["Iron", "Vit C"])
        case .none:
            return Recipe(title: "No conditions declared",
                          goal: "Stay active, sleep well, eat well.",
                          interventions: [])
        }
    }
}

/// Helper that returns a (reading, healthy) tuple per condition. Week 1
/// returns nil unless we have a hardcoded demo value; week 2 wires
/// HealthKit (`HKHealthStore`) for BP, glucose, HRV.
public enum CarePlanReadings {
    public static func reading(for condition: HealthCondition) -> (String, Bool)? {
        switch condition {
        case .hypertension:     return ("138/88", false) // mocked; HK in week 2
        case .diabetesT1, .diabetesT2: return ("A1C 6.1", false)
        case .heartCondition:   return ("HR 72", true)
        case .obesity:          return ("BMI 31", false)
        default: return nil
        }
    }
}
